packages/shared/src/hooks/contentPreference/useContentPreference.ts

```diff
@@ -0,0 +1,112 @@
+import { useMutation } from '@tanstack/react-query';
+import {
+  CONTENT_PREFERENCE_FOLLOW_MUTATION,
+  CONTENT_PREFERENCE_UNFOLLOW_MUTATION,
+  ContentPreferenceStatus,
+} from '../../graphql/contentPreference';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { gqlClient } from '../../graphql/common';
+import { PropsParameters } from '../../types';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { useToastNotification } from '../useToastNotification';
+import { ContentPreferenceMutation } from './types';
+
+export type UseContentPreference = {
+  follow: ContentPreferenceMutation;
+  unfollow: ContentPreferenceMutation;
+  subscribe: ContentPreferenceMutation;
+  unsubscribe: ContentPreferenceMutation;
+};
+
+export const useContentPreference = (): UseContentPreference => {
+  const { user } = useAuthContext();
+  const { displayToast } = useToastNotification();
+
+  const { mutateAsync: follow } = useMutation(
+    async ({
+      id,
+      entity,
+      entityName,
+    }: PropsParameters<UseContentPreference['follow']>) => {
+      await gqlClient.request(CONTENT_PREFERENCE_FOLLOW_MUTATION, {
+        id,
+        entity,
+        status: ContentPreferenceStatus.Follow,
+      });
+
+      displayToast(`✅ You are now following ${entityName}`);
```

Comment:
Oh I see, it's more of a display thing :)

Reply:
yes, just wanted to slim it down and make generic when we introduce other follow entities.